### DISCUSSION TITLE
ffmpeg RTMP to RtspSimpleServer to ffplay rtmp playback?

### DISCUSSION THREAD

**Discussion by Murmur** [bluenviron/mediamtx#790] [Questions] [answered]
As a testing scenario I had to create RTMP stream so I compiled this project to have `rtsp-simple-server.exe` on Windows 10. I was able to use the following commands (run on three separate DOS consoles).

Is this how RtspSimpleServer handles the RTMP protocol? I push to the given publishUrl and then playback the same url `rtmp://127.0.0.1:1935/stream1/stream1_720p` ? Is it possible to use a different publishUrl and playbackUrl?

```
rtsp-simple-server.exe

ffmpeg.exe -re -fflags +genpts+discardcorrupt -stream_loop -1 ^
  -i "c:/videos/test1.mp4"-c:v copy -c:a copy ^
  -f flv "rtmp://127.0.0.1:1935/stream1/stream1_720p flashver=FMLE/3.0\20(compatible;\20FMSc/1.0) live=true"

ffplay.exe "rtmp://127.0.0.1:1935/stream1/stream1_720p"
```

**Comment by aler9** [comment, marked as answer]
> I push to the given publishUrl and then playback the same url rtmp://127.0.0.1:1935/stream1/stream1_720p?

yes

> Is it possible to use a different publishUrl and playbackUrl?

no, the server supports multiple streams at once and uses unique URLs/paths to identify them. But you can use a separate username and a separate password for publishers and readers.

[a comment by an automated account is not shown]
